### PR TITLE
feat: add Buy a ticket pattern

### LINKED ISSUE #759: Add the Buy a ticket pattern which is used on the new homepage
## Documentation

# Buy a ticket
## What does it do?
* Helps users to enter the ticket purchase journey straight from the homepage

## When to use it?
* On the homepage

## How it works?
* This pattern directs users to the ticket finder service, where they can continue with purchasing a relevant ticket based on the ticket options they have selected

### PR DESCRIPTION
**Pull request by gldgrnt**
Fixes #759

### CODE REVIEW

**Review by jkmnch — CHANGES_REQUESTED** [review]
Selecting the Train or Tram buttons gives them a purple fill, instead of the standard active button style

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 1]
```diff
@@ -0,0 +1,112 @@
+{% macro wmndsBuyATicket() %}
+{# Imports #}
+{% from "wmnds/components/button/_button.njk" import wmndsButton %}
+{% from "wmnds/components/form-elements/dropdown/_dropdown.njk" import wmndsDropdown %}
+
+<div class="wmnds-buy-a-ticket">
+    <h2 class="wmnds-buy-a-ticket__title">Buy a ticket</h2>
+    {# Mode select #}
+    <div class="wmnds-m-b-md wmnds-p-b-xsm">
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:9]:
```suggestion - I will update form details after
    <form class="wmnds-m-b-md wmnds-p-b-xsm">
```

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                    wmndsButton({
+                        iconLeft: "modes-isolated-metro",
+                        contentText: "Tram",
+                        type: "mode",
+                        classes: "wmnds-btn--block"
+                    })
+                }}
+            </div>
+        </div>
+    </div>
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:43]:
```suggestion
    </form>
```

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                    wmndsButton({
+                        iconLeft: "modes-isolated-bus",
+                        contentText: "Bus",
+                        type: "mode",
+                        classes: "wmnds-btn--block wmnds-is--active"
+                    })
+                }}
+            </div>
+            <div class="wmnds-col-1-3">
+                {{
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:23]:
```suggestion
<input type="checkbox" class="wmnds-screenreaders-only" name="useTrain" />
                {{ 
```

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+{% from "wmnds/components/form-elements/dropdown/_dropdown.njk" import wmndsDropdown %}
+
+<div class="wmnds-buy-a-ticket">
+    <h2 class="wmnds-buy-a-ticket__title">Buy a ticket</h2>
+    {# Mode select #}
+    <div class="wmnds-m-b-md wmnds-p-b-xsm">
+        <label for="" class="wmnds-buy-a-ticket__label">Which modes of transport will you use?</label>
+        <div class="wmnds-grid wmnds-grid--spacing-3-sm wmnds-grid--spacing-sm-3-md">
+            <div class="wmnds-col-1-3">
+                {{
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:13]:
```suggestion
<input type="checkbox" class="wmnds-screenreaders-only" name="useBus" />
                {{ 
```

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                    wmndsButton({
+                        iconLeft: "modes-isolated-rail",
+                        contentText: "Train",
+                        type: "mode",
+                        classes: "wmnds-btn--block"
+                    })
+                }}
+            </div>
+            <div class="wmnds-col-1-3">
+                {{
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:33]:
```suggestion
                <input type="checkbox" class="wmnds-screenreaders-only" name="useTram" />
                {{ 
```

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                        classes: "wmnds-btn--block"
+                    })
+                }}
+            </div>
+        </div>
+    </div>
+    {# Traveller dropdown #}
+    {{
+        wmndsDropdown({
+            id: 'buyATicketTraveller',
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:47]: If wmndsDropdown accepts name for the input could you add **name: "PassengerType",**, please!

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                    contentText: 'Option 3',
+                    value: 'option3'
+                }
+            ]
+        })
+    }}
+    {# Duration dropdown #}
+    {{
+        wmndsDropdown({
+            id: "buyATicketDuration",
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:76]: same here: **name: "Duration",**

File: src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+                    value: 'option3'
+                }
+            ]
+        })
+    }}
+    {# Submit button #}
+    {{
+        wmndsButton({
+            iconRight: "general-chevron-right",
+            contentText: "Find tickets"
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/buy-a-ticket/_buy-a-ticket.njk:106]:
```suggestion
            contentText: "Find tickets",
            role: "submit"
```

**Review by catiarodriguescosta — CHANGES_REQUESTED** [review]
I left some suggestions that will turn my / Dayle's life easier. :)
But it is looking good! 👍

**Review by jkmnch — APPROVED** [review]

**Review by catiarodriguescosta — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**